mssql_rawquery_block.py

```diff
@@ -0,0 +1,50 @@
+from nio.properties import Property, VersionProperty, StringProperty
+from nio.block.terminals import output
+from .mssql_base import MSSQLBase
+from nio.block.mixins.enrich.enrich_signals import EnrichSignals
+
+@output('results', label='Results')
+@output('no_results', label='No Results')
+class MSSQLRawQuery(EnrichSignals, MSSQLBase):
+
+    version = VersionProperty("1.0.0")
+    query = StringProperty(title='Parameterized Query (use ? for any user-supplied values)', default='SELECT * FROM table where id=?', order=10)
+    parameters = Property(title='Substitution Parameters (As List, In Order)', default='{{[]}}', order=11)
+
+    def process_signals(self, signals):
+        if self.isConnecting:
+            self.logger.error('Connection already in progress. Dropping signals.')
+        else:
+            try:
+                cursor = self.cnxn.cursor()
+            except Exception as e:
+                self.disconnect()
+                self.connect()
+                cursor = self.cnxn.cursor()
+
+            output_signals = []
+
+            for signal in signals:
+                _query = self.query(signal)

```

Comment:
I don't think we can call `self.query` in any sort of dynamic signal-based context like this. There's just no way that's ever going to be up to snuff with security. If they are using the raw query block, they have to put a static query in here. The query parameters are where we can use an incoming signal's attributes.

Basically, this line needs to change to
```python
_query = self.query()
```

Reply:
The same is true for `self.table` elsewhere in this code. Table names should be fixed and not dependent on incoming signal attributes - the example @32bitkid gave earlier is the reason why.

Reply:
This has been changed to 
`_query = self.query()`